### PR TITLE
fix: Remove default registry reference in `info` cmd docs

### PR DESCRIPTION
**Pull request by Rustin170506**
### What does this PR try to resolve?

close https://github.com/rust-lang/cargo/issues/14810

Remove the registry description to reduce confusion, as the `spec` already explains local `Cargo.toml` inspection behavior.

### How should we test and review this PR?

The unit test has been updated.

### Additional information

r? @epage

### CODE REVIEW

File: src/doc/man/cargo-info.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ selected based on the Minimum Supported Rust Version (MSRV).
         cargo info serde@1.0.0
 3. Inspect the `serde` package form the local registry:
 
-        cargo info serde --registry my-registry
```
Comment (Rustin170506) [inline src/doc/man/cargo-info.md:63]: I'm not sure if it’s appropriate to include this change in the same commit. I’m fine with removing it or moving it to a separate commit if needed.

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: src/doc/man/cargo-info.md
Comment (weihanglo) [inline src/doc/man/cargo-info.md:1]: https://github.com/rust-lang/cargo/blob/0d730ac924ad35d2dca4b5cb208f356cee3312ae/src/doc/man/cargo-info.md?plain=1#L13


Should we adjust this description as well?

Comment (Rustin170506) [reply src/doc/man/cargo-info.md:1]: updated.

**Review by weihanglo — APPROVED** [review]
Thanks!
